sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2355,6 +2359,28 @@ class Dataset[T] private[sql](
   }
 
   /**
+   * Collect a Dataset to an ArrowRecordBatch.
+   *
+   * @group action
+   * @since 2.2.0
+   */
+  @DeveloperApi
+  def collectAsArrow(
+      allocator: RootAllocator = new RootAllocator(Long.MaxValue)): ArrowRecordBatch = {
```

Comment:
When we are able to adopt a streaming model, it might be useful for the Arrow payload data structure to be iterator based, so we can begin serving record batches to Python as soon as they begin arriving from the task executors